mongo/datadog_checks/mongo/dbm/utils.py

```diff
@@ -143,13 +144,23 @@ def should_explain_operation(
     return True
 
 
-def get_explain_plan(api_client, op: Optional[str], command: dict, dbname: str):
+def get_explain_plan(api_client, command: dict, dbname: str, op_duration: int, cursor_timeout: int) -> dict:
     dbname = command.pop("$db", dbname)
+    verbosity = get_explain_command_verbosity(op_duration, cursor_timeout)
     try:
         for key in EXPLAIN_COMMAND_EXCLUDE_KEYS:
             command.pop(key, None)
-        explain_plan = api_client[dbname].command("explain", command, verbosity="executionStats")
-        return format_explain_plan(explain_plan)
+        try:
+            explain_plan = api_client.explain_command(dbname, command, verbosity)
+            return format_explain_plan(explain_plan)
+        except (ExecutionTimeout, NetworkTimeout) as e:
+            # If the operation times out, we try one more time with a different verbosity
+            if verbosity != "queryPlanner":
```

Comment:
Do we need a debug log entry or a telemetry metric here to know when the timeouts happened?